internal/service/project/data_source_projects.go

```diff
@@ -171,17 +171,14 @@ func (d *ProjectsDS) Read(ctx context.Context, req datasource.ReadRequest, resp
 }
 
 func populateProjectsDataSourceModel(ctx context.Context, conn *matlas.Client, connV2 *admin.APIClient, stateModel *tfProjectsDSModel, projectsRes *matlas.Projects) error {
-	results := make([]*tfProjectDSModel, len(projectsRes.Results))
-
-	for i, project := range projectsRes.Results {
+	results := make([]*tfProjectDSModel, 0, len(projectsRes.Results))
+	for _, project := range projectsRes.Results {
 		atlasTeams, atlasLimits, atlasProjectSettings, err := getProjectPropsFromAPI(ctx, conn, connV2, project.ID)
-		if err != nil {
-			return fmt.Errorf("error while getting project properties for project %s: %v", project.ID, err.Error())
+		if err == nil { // if the project is still valid, e.g. could have just been deleted
```

Comment:
should we fail if `err != nil` ?

Reply:
no, that was the problem before.  Imagine we get the list of projects for the org, but then some of them is deleted.  It's not really a failure, so we just ignore it if it's deleted after it was in the list.   This was the reason for project flaky test.

Reply:
ok noted, but then I guess we're ok to ignore 5xx errors?
What I am trying to understand here is: if there is a 5xx error and we ignore that and we move on, will we get a different test behaviour that might hide some problem?

Reply:
I've got 500 sometimes when doing something in a project that is being closed, so we could get also a 500 and it doesn't mean a real error but that the project is closing.   So I think it's hard in this case to differentiate a "real" error from an error caused by the project being closed. Also chances are that if the project is in the list and it's closed we'll get the details without any problem

Reply:
Understand this an edge case that is likely only appearing  in our acceptance tests execution: fetching and creating/deleting project in parallel. Curious if a potential fix to this problem could be defining the data source test to run sequentially (`resource.Test(` over `resource.ParallelTest(`) to reduce the chances of this scenario. (just to keep in mind for other flaky tests that we might have)

Reply:
that would probably fix the tests, also to use the concurrency facility in GH actions so project acc and migration tests don't run in parallel.  But at the end I thought this is a realistic use case that can happen sometimes and we don't want the DS to fail if that happens, that's why I focused on improving the prod. code.